apps/outlook-rooms-addin/src/tests/map.service.spec.ts

```diff
@@ -0,0 +1,175 @@
+import { of } from 'rxjs';
+import { take } from 'rxjs/operators';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { BookModule } from '../app/rooms/book.module';
+import { CommonModule } from '@angular/common';
+import { BrowserModule } from '@angular/platform-browser';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import {
+    MatBottomSheetModule,
+    MatBottomSheet,
+} from '@angular/material/bottom-sheet';
+
+import { ComponentsModule, MapPinComponent } from '@placeos/components';
+
+import { RoomConfirmService } from '../app/rooms/room-confirm.service';
+import { RoomTileComponent } from '../app/rooms/room-tile/room-tile.component';
+import { mockRoomConfirmService, mockSpace } from './test-mocks';
+
+describe('MapService', () => {
+    let spectator: SpectatorService<MapService>;
+    const roomConfirmServiceStub = mockRoomConfirmService;
+
+    const createService = createServiceFactory({
+        service: MapService,
+        imports: [
+            ReactiveFormsModule,
+            FormsModule,
+            MatFormFieldModule,
+            BookModule,
+            CommonModule,
+            BrowserModule,
+            BrowserAnimationsModule,
+            MatBottomSheetModule,
+            ComponentsModule,
+        ],
+        providers: [
+            {
+                provide: MatBottomSheet,
+                useValue: {
+                    open: jest.fn((RoomTileComponent) => {}),
+                    afterDismissed: jest.fn(),
+                },
+            },
+            {
+                provide: RoomConfirmService,
+                useClass: roomConfirmServiceStub,
+            },
+        ],
+        declarations: [
+            MockComponent(RoomTileComponent),
+            MockComponent(MapPinComponent),
+        ],
+    });
+
+    beforeEach(() => {
+        spectator = createService();
+        jest.clearAllMocks();
+        jest.resetModules();
+        ngMocks.reset();
+    });
+
+    it('should create service', () => {
+        expect(spectator.service).toBeTruthy();
+    });
+
+    it('should return a selected space', async () => {
+        const room_service = spectator.inject(RoomConfirmService);
+        room_service.selected_space$ = of(mockSpace);
+
+        expect(JSON.stringify(spectator.service.selected_space$)).toBe(
+            JSON.stringify(room_service.selected_space$)
+        );
+    });
+    it('should return a list of map IDs with no duplicates', async () => {
+        const available_spaces = of([
+            mockSpace,
+            mockSpace,
+            mockSpace,
+            mockSpace,
+        ]);
+        let maps_list;
+        const map_list_spy = jest.spyOn(spectator.service, 'loadMap');
+        await spectator.service.locateSpaces(available_spaces);
+        spectator.service.maps_list$.subscribe((list) => {
```

Comment:
You should turn this into a promise to prevent the test being flaky.
e.g. `maps_list = await spectator.service.maps_list$.pipe(take(1).toPromise()`